src/functions-and-arrays.js

```diff
@@ -1,73 +1,434 @@
 // Iteration #1: Find the maximum
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num2 > num1) {
+    return num2;
+  } else {
+    return num1 && num2;
+  }
+}
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
+//const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words2 = [];
+function findLongestWord(array) {
+  var max = array[0];
+  //max = crocodile
+  if (array.length <= 0) {
+    return null;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      if (array[i].length > max.length) {
+        max = array[i];
+      }
+    }
+    return max;
+  }
+}
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(array) {
+  let sum = 0;
+
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
-const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+//const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+const numbersAvg = [];
+
+let sum = sumNumbers(numbersAvg);
+
+function averageNumbers(array, sum) {
+  if (array.length <= 0) {
+    return null;
+  } else {
+    return sum / array.length;
+  }
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
+
+let sumWordLength = function(array) {
+  let sum = 0;
+  if (array.length <= 0) {
+    return null;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      sum += array[i].length;
+    }
+    return sum;
+  }
+};
+
+//console.log(sumWordLength(wordsArr));
+
+let sum = sumWordLength(wordsArr);
+//sum = 53
+
+function averageWordLength(array, sum) {
```

Comment:
Same feedback as before iteration. The approach is very food but the execution could use some work. Instead of arguments and variables, just try to invoke the previous function inside this one with the argument array.